plugins/store/guest-js/index.ts

```diff
@@ -8,50 +8,294 @@ import { invoke, Resource } from '@tauri-apps/api/core'
 
 interface ChangePayload<T> {
   path: string
+  resourceId?: number
   key: string
-  value: T | null
+  value: T
+  exists: boolean
 }
 
 /**
  * Options to create a store
  */
 export type StoreOptions = {
   /**
-   * Auto save on modification with debounce duration in milliseconds
+   * Auto save on modification with debounce duration in milliseconds, it's 100ms by default, pass in `false` to disable it
    */
-  autoSave?: boolean
+  autoSave?: boolean | number
+  /**
+   * Name of a serialize function registered in the rust side plugin builder
+   */
+  serializeFnName?: string
+  /**
+   * Name of a deserialize function registered in the rust side plugin builder
+   */
+  deserializeFnName?: string
+}
+
+/**
+ * Create a new store.
+ *
+ * If the store already exists, its data will be overwritten.
+ *
+ * To load the store if it already exists you must use {@link load} instead.
+ *
+ * If the store is already loaded you must use {@link getStore} instead.
+ *
+ * @example
+ * ```typescript
+ * import { Store } from '@tauri-apps/api/store';
+ * const store = await Store.create('store.json');
+ * ```
+ *
+ * @param path Path to save the store in `app_data_dir`
+ * @param options Store configuration options
+ *
+ * @throws If a store at that path is already loaded
+ */
+export async function create(
+  path: string,
+  options?: StoreOptions
+): Promise<Store> {
+  return await Store.create(path, options)
 }
 
 /**
- * @param path: Path to save the store in `app_data_dir`
- * @param options: Store configuration options
+ * Create a new Store or load the existing store with the path.
+ *
+ * If the store is already loaded you must use {@link getStore} instead.
+ *
+ * @example
+ * ```typescript
+ * import { Store } from '@tauri-apps/api/store';
+ * let store = await Store.get('store.json');
+ * if (!store) {
+ *   store = await Store.load('store.json');
+ * }
+ * ```
+ *
+ * @param path Path to save the store in `app_data_dir`
+ * @param options Store configuration options
+ *
+ * @throws If a store at that path is already loaded
```

Comment:
Isn't this the same as `create`?

Reply:
I don't think so, `create` will create a new store and override old values. This will persist values from disk